pkg/server/plugin/datastore/sql/sql.go

```diff
@@ -919,10 +921,45 @@ func listRegistrationEntries(tx *gorm.DB,
 	}
 
 	return &datastore.ListRegistrationEntriesResponse{
-		Entries: entries,
+		Entries:    entries,
+		Pagination: req.Pagination,
 	}, nil
 }
 
+// applyPagination  add order limit and token to current query
+func applyPagination(p *datastore.Pagination, entryTx *gorm.DB) *gorm.DB {
+	return entryTx.Order("id asc").Limit(p.PageSize).Where("id > ?", p.Token)
+}
+
+// update pagination token based in last result in returned list
+func updatePaginationToken(p *datastore.Pagination, entries *[]RegisteredEntry) {
```

Comment:
it's seems weird to take a pointer to a slice...

Reply:
done